### PR TITLE
Add solution to Arguments Optional challenge

### PR DESCRIPTION
**Pull request by nikkiselev**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

### CODE REVIEW

File: guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/arguments-optional/index.md [hunk 1]
```diff
@@ -196,6 +196,25 @@ In the case that only one argument was passed, do not worry about how to prompt
...
+  if (typeof a !== 'number') {
+    return undefined;
+  }
+  const sum = b => (typeof b === 'number' ? a + b : undefined);
+  return typeof b === 'undefined' ? b => sum(b) : sum(b);
+}
+// test here
+addTogether(2,3);
+```
+![:rocket:](https://forum.freecodecamp.com/images/emoji/emoji_one/rocket.png?v=3 ":rocket:") <a href='https://repl.it/repls/SvelteWelloffLogin' target='_blank' rel='nofollow'>Run Code</a>
```
Comment (RandellDawson) [inline guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/arguments-optional/index.md]: @nikkiselev Please remove this line as we no longer accept repl.it links in challenge guide articles.
```suggestion
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
Hi @nikkiselev Your solution is concise.  The only improvement I would suggest outside of my request to remove the replt.it link, is to possible use variable names like `first` and `second` instead of `a` and `b` to make it a bit more readable.  We are discussing whether or not your solution should just replace the existing Advanced solution instead of adding another one.  We will get back to you soon regarding that decision.

Great work!

**Comment by nikkiselev** [comment]
@RandellDawson , I pushed changes.
Thanks.

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
@nikkiselev We decided to replace the existing Intermediate solution with yours.

It will be  a few more days until we can merge this one, as we are awaiting another PR with changes to this same file to be merged first.

File: guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/arguments-optional/index.md [hunk 2]
```diff
@@ -105,65 +104,31 @@ In the case that only one argument was passed, do not worry about how to prompt
...
-    // test here
-    addTogether(2,3);
+function addTogether(first, second) {
+  if (typeof first !== 'number') {
+    return undefined;
+  }
+  const sum = second => (typeof second === 'number' ? first + second : undefined);
+  return typeof second === 'undefined' ? second => sum(second) : sum(second);
+}
+// test here
```
Comment (mwangi-njuguna) [inline guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/arguments-optional/index.md]: What is supposed to happen as per the example when the typeof **second** is **null**(_typeof null === 'object'_)?

Comment (nikkiselev) [reply guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/arguments-optional/index.md]: Hi @mwangi-njuguna , it returns `undefined` as required `If either argument isn't a valid number, return undefined.` in the task.

**Comment by RandellDawson** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 🎉

**Pull request merged**